.github/workflows/build-microzig.yml

```diff
@@ -0,0 +1,37 @@
+name: Build Microzig
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - .github/**
+      - build.zig*
+      - tools/**
+      - core/**
+      - port/**
+      - examples/**
+  pull_request:
+    branches: [main]
+    paths:
+      - build.zig*
+      - tools/**
+      - core/**
+      - port/**
+      - examples/**
```

Comment:
We have to always run the MicroZig CI as changes on regz, uf2, ... can also break MicroZig

Reply:
This runs on changes for regz and uf2 as well, but based on our conversations I have removed the path filters.